### PR TITLE
Enable profiling in perf piplines.

### PR DESCRIPTION
**Pull request by gearama**
To enable profiling , along with chnges in the automation the changes to the pipelines are required 
1. Installl valgrind -- profiling tool for linux and cpp 
2. add option to run profiling, 
3. pass it to automation 

# Pull Request Checklist

Please leverage this checklist as a reminder to address commonly occurring feedback when submitting a pull request to make sure your PR can be reviewed quickly:

See the detailed list in the [contributing guide](https://github.com/Azure/azure-sdk-for-cpp/blob/main/CONTRIBUTING.md#pull-requests).

- [ ] [C++ Guidelines](https://azure.github.io/azure-sdk/cpp_introduction.html)
- [ ] Doxygen docs
- [ ] Unit tests
- [ ] No unwanted commits/changes
- [ ] Descriptive title/description
  - [ ] PR is single purpose
  - [ ] Related issue listed
- [ ] Comments in source
- [ ] No typos
- [ ] Update changelog
- [ ] Not work-in-progress
- [ ] External references or docs updated
- [ ] Self review of PR done
- [ ] Any breaking changes?

### CODE REVIEW

**Comment by gearama** [comment]
ignore cspell for the moment, Daniel is looking into why it failed, as there are no real failures, it seems to have grabbed the wrong version of cspell for some reason

**Comment by ahsonkhan** [comment]
> ignore cspell when reviewing for the moment, Daniel is looking into why it failed, as there are no real failures, it seems to have grabbed the wrong version of cspell for some reason

I am fixing [that](https://github.com/Azure/azure-sdk-for-cpp/issues/4395) in https://github.com/Azure/azure-sdk-for-cpp/pull/4393 but am hitting another infra issue, there:
https://github.com/Azure/azure-sdk-for-cpp/issues/4397

File: eng/pipelines/templates/jobs/perf.yml [hunk 1]
```diff
@@ -31,11 +34,20 @@ extends:
     - template: /eng/pipelines/templates/variables/globals.yml
     Language: Cpp
     LanguageVersion: 'N/A'
-    InstallLanguageSteps: ${{ parameters.InstallLanguageSteps }}
+    InstallLanguageSteps:
+      - pwsh: |
+          Write-Host "##vso[task.setvariable variable=VCPKG_BINARY_SOURCES_SECRET;issecret=true;]clear;x-azblob,https://cppvcpkgcache.blob.core.windows.net/public-vcpkg-container,,read"
+        displayName: Set Vcpkg Variables
+      - ${{ if parameters.Profile }}:
```
Comment (LarryOsterman) [inline eng/pipelines/templates/jobs/perf.yml:41]: Silly question: When is this set to "true"?

Comment (gearama) [reply eng/pipelines/templates/jobs/perf.yml:41]: when you select it in the pipeline run :) 
it's presented as a check box as one of the options of the run , the idea is you want to run profiling for the perf tests , this is how

**Review by LarryOsterman — APPROVED** [review]
Looks reasonable to me.

**Comment by danieljurek** [comment]
The spell checker issue is fixed. Rebase this PR to get a more successful run of the spell checker.

**Review by mikeharder — APPROVED** [review]

File: eng/pipelines/templates/jobs/perf.yml [hunk 2]
```diff
@@ -31,11 +34,20 @@ extends:
...
+        - task: Bash@3
+          inputs:
+            targetType: 'inline'
+            script: sudo apt install valgrind -y
```
Comment (mikeharder) [inline eng/pipelines/templates/jobs/perf.yml:45]:
```suggestion
            script: sudo apt install valgrind -y
            displayName: Install valgrind
```

Comment (mikeharder) [reply eng/pipelines/templates/jobs/perf.yml:45]: Please add a `displayName`, otherwise this just appears as `Bash` in the pipeline step list

**Review by mikeharder — CHANGES_REQUESTED** [review]

**Review by mikeharder — APPROVED** [review]

**Pull request merged**